lib/net/nforceif/include/lwipopts.h

```diff
@@ -47,6 +47,24 @@
 #define LWIP_ERRNO_STDINCLUDE 1
 #define LWIP_COMPAT_MUTEX_ALLOWED
 
+
+#define LWIP_HOOK_DHCP_APPEND_OPTIONS(netif, dhcp, state, msg, msg_type, options_len_ptr) \
+    *options_len_ptr = dhcp_option(*options_len_ptr, msg->options, DHCP_OPTION_CLIENT_ID, 7); \
+    *options_len_ptr = dhcp_option_byte(*options_len_ptr, msg->options, 0x01); \
+    for (int _i = 0; _i < 6; _i++) { \
+        *options_len_ptr = dhcp_option_byte(*options_len_ptr, msg->options, netif->hwaddr[_i]); \
+    } \
+    \
+    *options_len_ptr = dhcp_option(*options_len_ptr, msg->options, DHCP_OPTION_US, 8); \
+    *options_len_ptr = dhcp_option_byte(*options_len_ptr, msg->options, 'X'); \
+    *options_len_ptr = dhcp_option_byte(*options_len_ptr, msg->options, 'B'); \
+    *options_len_ptr = dhcp_option_byte(*options_len_ptr, msg->options, 'O'); \
+    *options_len_ptr = dhcp_option_byte(*options_len_ptr, msg->options, 'X'); \
+    *options_len_ptr = dhcp_option_byte(*options_len_ptr, msg->options, ' '); \
+    *options_len_ptr = dhcp_option_byte(*options_len_ptr, msg->options, '1'); \
+    *options_len_ptr = dhcp_option_byte(*options_len_ptr, msg->options, '.'); \
+    *options_len_ptr = dhcp_option_byte(*options_len_ptr, msg->options, '0');
```

Comment:
I didn't check the upstream file again, but I believe I'd rather add all of this where it is in the upstream `opt.h`, including the comment. We should try to minimize our diff so we can react when upstream changes something dumb (like macro-parameter order).

Reply:
I did that now. I also added the comment explaining the additions, which I forgot to commit earlier.

Reply:
nit: That comment from the edit could have been part of the code-comment